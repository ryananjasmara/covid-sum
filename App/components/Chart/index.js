/* eslint-disable react-hooks/exhaustive-deps */
import React from 'react';
import {StyleSheet, View, Text, Dimensions} from 'react-native';
import {Picker} from '@react-native-community/picker';
import {LineChart} from 'react-native-chart-kit';
import {thousandSeparator} from '../../helpers';
import WORDS from '../../configs/words';

const Chart = ({data}) => {
  const [selectedCategories, setSelectedCategories] = React.useState(
    WORDS.CONFIRMED,
  );
  const [selectedCountries, setSelectedCountries] = React.useState(
    data.confirmed.locations[0].country,
  );
  const [countriesList, setCountriesList] = React.useState(
    data.confirmed.locations,
  );
  const [chartData, setChartData] = React.useState(data.confirmed.locations[0]);

  React.useEffect(() => {
    // setCountriesList(data[WORDS.CONFIRMED.toLowerCase()].locations);
    switch (selectedCategories) {
      case WORDS.CONFIRMED:
        setCountriesList(data.confirmed.locations);
        break;
      case WORDS.DEATHS:
        setCountriesList(data.deaths.locations);
        break;
      case WORDS.RECOVERED:
        setCountriesList(data.recovered.locations);
        break;
    }
  }, [selectedCategories]);

  React.useEffect(() => {
    const specificCountry = countriesList.find((item) => {
      return item.country === selectedCountries;
    });
    setChartData(specificCountry);
  }, [selectedCountries, countriesList]);

  function renderCategoriesPicker() {
    return (
      <Picker
        selectedValue={selectedCategories}
        style={styles.categoriesPicker}
        onValueChange={(itemValue) => {
          setSelectedCategories(itemValue);
        }}>
        <Picker.Item key label={WORDS.CONFIRMED} value={WORDS.CONFIRMED} />
        <Picker.Item label={WORDS.DEATHS} value={WORDS.DEATHS} />
        <Picker.Item label={WORDS.RECOVERED} value={WORDS.RECOVERED} />
      </Picker>
    );
  }

  function renderCountryPicker() {
    return (
      <Picker
        selectedValue={selectedCountries}
        style={styles.categoriesPicker}
        onValueChange={(itemValue) => {
          setSelectedCountries(itemValue);
        }}>
        {countriesList.map((item, index) => {
          return (
            <Picker.Item
              key={index}
              label={item.country}
              value={item.country}
            />
          );
        })}
      </Picker>
    );
  }

  function getLabelData(chartData) {
    const label = Object.keys(chartData.history);
    label.reverse();
    const slicedLabel = label.slice(0, 5);
    slicedLabel.reverse();
    return slicedLabel;
  }

  function getLabelValue(chartData) {
    const value = Object.values(chartData.history);
    value.reverse();
    const slicedValue = value.slice(0, 5);
    slicedValue.reverse();
    return slicedValue;
  }

  return (
    <View style={styles.container}>
      <View style={styles.headerTitleContainer}>
        <Text style={styles.headerTitleText}>{WORDS.CHART_HEADER}</Text>
      </View>
      <View style={styles.pickerContainer}>
        {renderCountryPicker()}
        {renderCategoriesPicker()}
      </View>
      <Text style={styles.latestCountText}>{`Total ${thousandSeparator(
        chartData.latest,
      )} ${selectedCategories} case's in ${selectedCountries}`}</Text>
      <LineChart
        data={{
          labels: getLabelData(chartData),
          datasets: [
            {
              data: getLabelValue(chartData),
            },
          ],
        }}
        width={Dimensions.get('window').width}
        height={220}
        yAxisInterval={1}
        chartConfig={{
          backgroundColor: '#e26a00',
          backgroundGradientFrom: '#fb8c00',
          backgroundGradientTo: '#ffa726',
          decimalPlaces: 2,
          color: (opacity = 1) => `rgba(255, 255, 255, ${opacity})`,
          labelColor: (opacity = 1) => `rgba(255, 255, 255, ${opacity})`,
          style: styles.chartConfigStyle,
          propsForDots: {
            r: '6',
            strokeWidth: '2',
            stroke: '#ffa726',
          },
        }}
        bezier
        style={styles.chart}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
  },
  chart: {
    borderRadius: 5,
    margin: 10,
  },
  chartConfigStyle: {
    borderRadius: 16,
  },
  headerTitleContainer: {
    alignItems: 'center',
  },
  headerTitleText: {
    fontSize: 18,
    fontWeight: 'bold',
  },
  pickerContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  categoriesPicker: {
    height: 50,
    width: 150,
  },
  latestCountText: {
    fontSize: 12,
  },
});

export default Chart;
