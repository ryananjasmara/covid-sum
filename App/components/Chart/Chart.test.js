import React from 'react';
import renderer from 'react-test-renderer';
import Chart from './index';

describe('Chart Component', () => {
  const dummies = {
    country: 'Afghanistan',
    country_code: 'AF',
    province: '',
    coordinates: {
      lat: '33.0',
      long: '65.0',
    },
    history: {
      '1/22/20': 0,
      '1/23/20': 0,
      '1/24/20': 0,
      '1/25/20': 0,
      '1/26/20': 0,
      '1/27/20': 0,
      '1/28/20': 0,
      '1/29/20': 0,
      '1/30/20': 0,
      '1/31/20': 0,
      '2/1/20': 0,
      '2/2/20': 0,
      '2/3/20': 0,
      '2/4/20': 0,
      '2/5/20': 0,
      '2/6/20': 0,
      '2/7/20': 0,
      '2/8/20': 0,
      '2/9/20': 0,
      '2/10/20': 0,
      '2/11/20': 0,
      '2/12/20': 0,
      '2/13/20': 0,
      '2/14/20': 0,
      '2/15/20': 0,
      '2/16/20': 0,
      '2/17/20': 0,
      '2/18/20': 0,
      '2/19/20': 0,
      '2/20/20': 0,
      '2/21/20': 0,
      '2/22/20': 0,
      '2/23/20': 0,
      '2/24/20': 1,
      '2/25/20': 1,
      '2/26/20': 1,
      '2/27/20': 1,
      '2/28/20': 1,
      '2/29/20': 1,
      '3/1/20': 1,
      '3/2/20': 1,
      '3/3/20': 1,
      '3/4/20': 1,
      '3/5/20': 1,
      '3/6/20': 1,
      '3/7/20': 1,
      '3/8/20': 4,
      '3/9/20': 4,
      '3/10/20': 5,
      '3/11/20': 7,
      '3/12/20': 7,
      '3/13/20': 7,
      '3/14/20': 11,
      '3/15/20': 16,
      '3/16/20': 21,
      '3/17/20': 22,
      '3/18/20': 22,
      '3/19/20': 22,
      '3/20/20': 24,
      '3/21/20': 24,
      '3/22/20': 40,
      '3/23/20': 40,
      '3/24/20': 74,
      '3/25/20': 84,
      '3/26/20': 94,
      '3/27/20': 110,
      '3/28/20': 110,
      '3/29/20': 120,
      '3/30/20': 170,
      '3/31/20': 174,
      '4/1/20': 237,
      '4/2/20': 273,
      '4/3/20': 281,
      '4/4/20': 299,
      '4/5/20': 349,
      '4/6/20': 367,
      '4/7/20': 423,
      '4/8/20': 444,
      '4/9/20': 484,
      '4/10/20': 521,
      '4/11/20': 555,
      '4/12/20': 607,
      '4/13/20': 665,
      '4/14/20': 714,
      '4/15/20': 784,
      '4/16/20': 840,
      '4/17/20': 906,
      '4/18/20': 933,
      '4/19/20': 996,
      '4/20/20': 1026,
      '4/21/20': 1092,
      '4/22/20': 1176,
      '4/23/20': 1279,
      '4/24/20': 1351,
      '4/25/20': 1463,
      '4/26/20': 1531,
      '4/27/20': 1703,
      '4/28/20': 1828,
    },
    latest: 1828,
  };
  const initialProps = {
    data: {
      confirmed: {
        locations: [dummies],
      },
      deaths: {
        locations: [dummies],
      },
      recovered: {
        locations: [dummies],
      },
    },
  };

  test('renders with data', () => {
    const tree = renderer.create(<Chart {...initialProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders without data', () => {
    const tree = renderer.create(<Chart />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
