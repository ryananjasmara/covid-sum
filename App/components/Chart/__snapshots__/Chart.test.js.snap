// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart Component renders with data 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      Graph Per Countries
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        Object {
          "height": 50,
          "width": 150,
        }
      }
    >
      <RNCPicker
        items={
          Array [
            Object {
              "label": "Afghanistan",
              "textColor": undefined,
              "value": "Afghanistan",
            },
          ]
        }
        onChange={[Function]}
        selectedIndex={0}
        style={
          Array [
            Object {
              "height": 216,
            },
            undefined,
          ]
        }
      />
    </View>
    <View
      style={
        Object {
          "height": 50,
          "width": 150,
        }
      }
    >
      <RNCPicker
        items={
          Array [
            Object {
              "label": "Confirmed",
              "textColor": undefined,
              "value": "Confirmed",
            },
            Object {
              "label": "Deaths",
              "textColor": undefined,
              "value": "Deaths",
            },
            Object {
              "label": "Recovered",
              "textColor": undefined,
              "value": "Recovered",
            },
          ]
        }
        onChange={[Function]}
        selectedIndex={0}
        style={
          Array [
            Object {
              "height": 216,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
  <Text
    style={
      Object {
        "fontSize": 12,
      }
    }
  >
    Total 1.828 Confirmed case's in Afghanistan
  </Text>
  <View
    style={
      Object {
        "borderRadius": 5,
        "margin": 10,
      }
    }
  >
    <RNSVGSvgView
      bbHeight={220}
      bbWidth={730}
      focusable={false}
      height={220}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": 220,
            "width": 730,
          },
        ]
      }
      width={730}
    >
      <RNSVGGroup>
        <RNSVGRect
          fill={
            Array [
              1,
              "backgroundGradient",
            ]
          }
          fillOpacity={1}
          height={220}
          propList={
            Array [
              "fill",
              "fillOpacity",
            ]
          }
          rx={5}
          ry={5}
          width="100%"
          x={0}
          y={0}
        />
        <RNSVGGroup
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
        >
          <RNSVGDefs>
            <RNSVGLinearGradient
              gradient={
                Array [
                  0,
                  -291840,
                  1,
                  -22746,
                ]
              }
              gradientTransform={null}
              gradientUnits={0}
              name="backgroundGradient"
              x1="0"
              x2={750}
              y1={220}
              y2={0}
            />
            <RNSVGLinearGradient
              gradient={
                Array [
                  0,
                  452984831,
                  1,
                  16777215,
                ]
              }
              gradientTransform={null}
              gradientUnits={0}
              name="fillShadowGradient"
              x1={0}
              x2={0}
              y1={0}
              y2={220}
            />
          </RNSVGDefs>
          <RNSVGGroup>
            <RNSVGLine
              propList={
                Array [
                  "stroke",
                  "strokeWidth",
                  "strokeDasharray",
                ]
              }
              stroke={872415231}
              strokeDasharray={
                Array [
                  "5",
                  "10",
                ]
              }
              strokeWidth={1}
              x1={64}
              x2={750}
              y1={16}
              y2={16}
            />
            <RNSVGLine
              propList={
                Array [
                  "stroke",
                  "strokeWidth",
                  "strokeDasharray",
                ]
              }
              stroke={872415231}
              strokeDasharray={
                Array [
                  "5",
                  "10",
                ]
              }
              strokeWidth={1}
              x1={64}
              x2={750}
              y1={57.25}
              y2={57.25}
            />
            <RNSVGLine
              propList={
                Array [
                  "stroke",
                  "strokeWidth",
                  "strokeDasharray",
                ]
              }
              stroke={872415231}
              strokeDasharray={
                Array [
                  "5",
                  "10",
                ]
              }
              strokeWidth={1}
              x1={64}
              x2={750}
              y1={98.5}
              y2={98.5}
            />
            <RNSVGLine
              propList={
                Array [
                  "stroke",
                  "strokeWidth",
                  "strokeDasharray",
                ]
              }
              stroke={872415231}
              strokeDasharray={
                Array [
                  "5",
                  "10",
                ]
              }
              strokeWidth={1}
              x1={64}
              x2={750}
              y1={139.75}
              y2={139.75}
            />
            <RNSVGLine
              propList={
                Array [
                  "stroke",
                  "strokeWidth",
                  "strokeDasharray",
                ]
              }
              stroke={872415231}
              strokeDasharray={
                Array [
                  "5",
                  "10",
                ]
              }
              strokeWidth={1}
              x1={64}
              x2={750}
              y1={181}
              y2={181}
            />
          </RNSVGGroup>
          <RNSVGGroup>
            <RNSVGText
              content={null}
              dx={Array []}
              dy={Array []}
              fill={3439329279}
              font={
                Object {
                  "fontSize": 12,
                  "textAnchor": "end",
                }
              }
              matrix={
                Array [
                  1,
                  0,
                  -0,
                  1,
                  0,
                  0,
                ]
              }
              propList={
                Array [
                  "fill",
                ]
              }
              rotate={Array []}
              x={
                Array [
                  52,
                ]
              }
              y={
                Array [
                  181,
                ]
              }
            >
              <RNSVGTSpan
                content="1351.00"
                dx={Array []}
                dy={Array []}
                font={Object {}}
                rotate={Array []}
                x={Array []}
                y={Array []}
              />
            </RNSVGText>
            <RNSVGText
              content={null}
              dx={Array []}
              dy={Array []}
              fill={3439329279}
              font={
                Object {
                  "fontSize": 12,
                  "textAnchor": "end",
                }
              }
              matrix={
                Array [
                  1,
                  0,
                  -0,
                  1,
                  0,
                  0,
                ]
              }
              propList={
                Array [
                  "fill",
                ]
              }
              rotate={Array []}
              x={
                Array [
                  52,
                ]
              }
              y={
                Array [
                  139.75,
                ]
              }
            >
              <RNSVGTSpan
                content="1470.25"
                dx={Array []}
                dy={Array []}
                font={Object {}}
                rotate={Array []}
                x={Array []}
                y={Array []}
              />
            </RNSVGText>
            <RNSVGText
              content={null}
              dx={Array []}
              dy={Array []}
              fill={3439329279}
              font={
                Object {
                  "fontSize": 12,
                  "textAnchor": "end",
                }
              }
              matrix={
                Array [
                  1,
                  0,
                  -0,
                  1,
                  0,
                  0,
                ]
              }
              propList={
                Array [
                  "fill",
                ]
              }
              rotate={Array []}
              x={
                Array [
                  52,
                ]
              }
              y={
                Array [
                  98.5,
                ]
              }
            >
              <RNSVGTSpan
                content="1589.50"
                dx={Array []}
                dy={Array []}
                font={Object {}}
                rotate={Array []}
                x={Array []}
                y={Array []}
              />
            </RNSVGText>
            <RNSVGText
              content={null}
              dx={Array []}
              dy={Array []}
              fill={3439329279}
              font={
                Object {
                  "fontSize": 12,
                  "textAnchor": "end",
                }
              }
              matrix={
                Array [
                  1,
                  0,
                  -0,
                  1,
                  0,
                  0,
                ]
              }
              propList={
                Array [
                  "fill",
                ]
              }
              rotate={Array []}
              x={
                Array [
                  52,
                ]
              }
              y={
                Array [
                  57.25,
                ]
              }
            >
              <RNSVGTSpan
                content="1708.75"
                dx={Array []}
                dy={Array []}
                font={Object {}}
                rotate={Array []}
                x={Array []}
                y={Array []}
              />
            </RNSVGText>
            <RNSVGText
              content={null}
              dx={Array []}
              dy={Array []}
              fill={3439329279}
              font={
                Object {
                  "fontSize": 12,
                  "textAnchor": "end",
                }
              }
              matrix={
                Array [
                  1,
                  0,
                  -0,
                  1,
                  0,
                  0,
                ]
              }
              propList={
                Array [
                  "fill",
                ]
              }
              rotate={Array []}
              x={
                Array [
                  52,
                ]
              }
              y={
                Array [
                  16,
                ]
              }
            >
              <RNSVGTSpan
                content="1828.00"
                dx={Array []}
                dy={Array []}
                font={Object {}}
                rotate={Array []}
                x={Array []}
                y={Array []}
              />
            </RNSVGText>
          </RNSVGGroup>
          <RNSVGGroup>
            <RNSVGLine
              propList={
                Array [
                  "stroke",
                  "strokeWidth",
                  "strokeDasharray",
                ]
              }
              stroke={872415231}
              strokeDasharray={
                Array [
                  "5",
                  "10",
                ]
              }
              strokeWidth={1}
              x1={64}
              x2={64}
              y1={0}
              y2={181}
            />
            <RNSVGLine
              propList={
                Array [
                  "stroke",
                  "strokeWidth",
                  "strokeDasharray",
                ]
              }
              stroke={872415231}
              strokeDasharray={
                Array [
                  "5",
                  "10",
                ]
              }
              strokeWidth={1}
              x1={201}
              x2={201}
              y1={0}
              y2={181}
            />
            <RNSVGLine
              propList={
                Array [
                  "stroke",
                  "strokeWidth",
                  "strokeDasharray",
                ]
              }
              stroke={872415231}
              strokeDasharray={
                Array [
                  "5",
                  "10",
                ]
              }
              strokeWidth={1}
              x1={338}
              x2={338}
              y1={0}
              y2={181}
            />
            <RNSVGLine
              propList={
                Array [
                  "stroke",
                  "strokeWidth",
                  "strokeDasharray",
                ]
              }
              stroke={872415231}
              strokeDasharray={
                Array [
                  "5",
                  "10",
                ]
              }
              strokeWidth={1}
              x1={475}
              x2={475}
              y1={0}
              y2={181}
            />
            <RNSVGLine
              propList={
                Array [
                  "stroke",
                  "strokeWidth",
                  "strokeDasharray",
                ]
              }
              stroke={872415231}
              strokeDasharray={
                Array [
                  "5",
                  "10",
                ]
              }
              strokeWidth={1}
              x1={612}
              x2={612}
              y1={0}
              y2={181}
            />
          </RNSVGGroup>
          <RNSVGGroup>
            <RNSVGText
              content={null}
              dx={Array []}
              dy={Array []}
              fill={3439329279}
              font={
                Object {
                  "fontSize": 12,
                  "textAnchor": "middle",
                }
              }
              matrix={
                Array [
                  1,
                  0,
                  -0,
                  1,
                  0,
                  0,
                ]
              }
              propList={
                Array [
                  "fill",
                ]
              }
              rotate={Array []}
              x={
                Array [
                  64,
                ]
              }
              y={
                Array [
                  205,
                ]
              }
            >
              <RNSVGTSpan
                content="4/24/20"
                dx={Array []}
                dy={Array []}
                font={Object {}}
                rotate={Array []}
                x={Array []}
                y={Array []}
              />
            </RNSVGText>
            <RNSVGText
              content={null}
              dx={Array []}
              dy={Array []}
              fill={3439329279}
              font={
                Object {
                  "fontSize": 12,
                  "textAnchor": "middle",
                }
              }
              matrix={
                Array [
                  1,
                  0,
                  -0,
                  1,
                  0,
                  0,
                ]
              }
              propList={
                Array [
                  "fill",
                ]
              }
              rotate={Array []}
              x={
                Array [
                  201.2,
                ]
              }
              y={
                Array [
                  205,
                ]
              }
            >
              <RNSVGTSpan
                content="4/25/20"
                dx={Array []}
                dy={Array []}
                font={Object {}}
                rotate={Array []}
                x={Array []}
                y={Array []}
              />
            </RNSVGText>
            <RNSVGText
              content={null}
              dx={Array []}
              dy={Array []}
              fill={3439329279}
              font={
                Object {
                  "fontSize": 12,
                  "textAnchor": "middle",
                }
              }
              matrix={
                Array [
                  1,
                  0,
                  -0,
                  1,
                  0,
                  0,
                ]
              }
              propList={
                Array [
                  "fill",
                ]
              }
              rotate={Array []}
              x={
                Array [
                  338.4,
                ]
              }
              y={
                Array [
                  205,
                ]
              }
            >
              <RNSVGTSpan
                content="4/26/20"
                dx={Array []}
                dy={Array []}
                font={Object {}}
                rotate={Array []}
                x={Array []}
                y={Array []}
              />
            </RNSVGText>
            <RNSVGText
              content={null}
              dx={Array []}
              dy={Array []}
              fill={3439329279}
              font={
                Object {
                  "fontSize": 12,
                  "textAnchor": "middle",
                }
              }
              matrix={
                Array [
                  1,
                  0,
                  -0,
                  1,
                  0,
                  0,
                ]
              }
              propList={
                Array [
                  "fill",
                ]
              }
              rotate={Array []}
              x={
                Array [
                  475.59999999999997,
                ]
              }
              y={
                Array [
                  205,
                ]
              }
            >
              <RNSVGTSpan
                content="4/27/20"
                dx={Array []}
                dy={Array []}
                font={Object {}}
                rotate={Array []}
                x={Array []}
                y={Array []}
              />
            </RNSVGText>
            <RNSVGText
              content={null}
              dx={Array []}
              dy={Array []}
              fill={3439329279}
              font={
                Object {
                  "fontSize": 12,
                  "textAnchor": "middle",
                }
              }
              matrix={
                Array [
                  1,
                  0,
                  -0,
                  1,
                  0,
                  0,
                ]
              }
              propList={
                Array [
                  "fill",
                ]
              }
              rotate={Array []}
              x={
                Array [
                  612.8,
                ]
              }
              y={
                Array [
                  205,
                ]
              }
            >
              <RNSVGTSpan
                content="4/28/20"
                dx={Array []}
                dy={Array []}
                font={Object {}}
                rotate={Array []}
                x={Array []}
                y={Array []}
              />
            </RNSVGText>
          </RNSVGGroup>
          <RNSVGGroup>
            <RNSVGPath
              d="M64,181 Q 98.25, 181, 132.5, 161.5 Q 166.75, 142, 201, 142 Q 235.25, 142, 269.5, 130 Q 303.75, 118, 338, 118 Q 372.25, 118, 406.5, 88.5 Q 440.75, 59, 475, 59 Q 509.25, 59, 543.5, 37.5 Q 577.75, 16, 612, 16"
              fill={null}
              propList={
                Array [
                  "fill",
                  "stroke",
                  "strokeWidth",
                ]
              }
              stroke={872415231}
              strokeWidth={3}
            />
          </RNSVGGroup>
          <RNSVGGroup>
            <RNSVGPath
              d="M64,181 Q 98.25, 181, 132.5, 161.5 Q 166.75, 142, 201, 142 Q 235.25, 142, 269.5, 130 Q 303.75, 118, 338, 118 Q 372.25, 118, 406.5, 88.5 Q 440.75, 59, 475, 59 Q 509.25, 59, 543.5, 37.5 Q 577.75, 16, 612, 16 L612.8,181 L64,181 Z"
              fill={
                Array [
                  1,
                  "fillShadowGradient",
                ]
              }
              propList={
                Array [
                  "fill",
                  "strokeWidth",
                ]
              }
              strokeWidth={0}
            />
          </RNSVGGroup>
          <RNSVGGroup>
            <RNSVGCircle
              cx={64}
              cy={181}
              fill={3875536895}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              propList={
                Array [
                  "fill",
                  "stroke",
                  "strokeWidth",
                ]
              }
              r="6"
              responsible={true}
              stroke={4294944550}
              strokeWidth="2"
            />
            <RNSVGCircle
              cx={64}
              cy={181}
              fill={4294967295}
              fillOpacity={0}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              propList={
                Array [
                  "fill",
                  "fillOpacity",
                ]
              }
              r="14"
              responsible={true}
            />
            <RNSVGCircle
              cx={201.2}
              cy={142.2578616352201}
              fill={3875536895}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              propList={
                Array [
                  "fill",
                  "stroke",
                  "strokeWidth",
                ]
              }
              r="6"
              responsible={true}
              stroke={4294944550}
              strokeWidth="2"
            />
            <RNSVGCircle
              cx={201.2}
              cy={142.2578616352201}
              fill={4294967295}
              fillOpacity={0}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              propList={
                Array [
                  "fill",
                  "fillOpacity",
                ]
              }
              r="14"
              responsible={true}
            />
            <RNSVGCircle
              cx={338.4}
              cy={118.73584905660377}
              fill={3875536895}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              propList={
                Array [
                  "fill",
                  "stroke",
                  "strokeWidth",
                ]
              }
              r="6"
              responsible={true}
              stroke={4294944550}
              strokeWidth="2"
            />
            <RNSVGCircle
              cx={338.4}
              cy={118.73584905660377}
              fill={4294967295}
              fillOpacity={0}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              propList={
                Array [
                  "fill",
                  "fillOpacity",
                ]
              }
              r="14"
              responsible={true}
            />
            <RNSVGCircle
              cx={475.6}
              cy={59.23899371069182}
              fill={3875536895}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              propList={
                Array [
                  "fill",
                  "stroke",
                  "strokeWidth",
                ]
              }
              r="6"
              responsible={true}
              stroke={4294944550}
              strokeWidth="2"
            />
            <RNSVGCircle
              cx={475.6}
              cy={59.23899371069182}
              fill={4294967295}
              fillOpacity={0}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              propList={
                Array [
                  "fill",
                  "fillOpacity",
                ]
              }
              r="14"
              responsible={true}
            />
            <RNSVGCircle
              cx={612.8}
              cy={16}
              fill={3875536895}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              propList={
                Array [
                  "fill",
                  "stroke",
                  "strokeWidth",
                ]
              }
              r="6"
              responsible={true}
              stroke={4294944550}
              strokeWidth="2"
            />
            <RNSVGCircle
              cx={612.8}
              cy={16}
              fill={4294967295}
              fillOpacity={0}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              propList={
                Array [
                  "fill",
                  "fillOpacity",
                ]
              }
              r="14"
              responsible={true}
            />
          </RNSVGGroup>
          <RNSVGGroup />
          <RNSVGGroup />
        </RNSVGGroup>
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;

exports[`Chart Component renders without data 1`] = `null`;

exports[`renders correctly 1`] = `null`;
